packages/editor/src/components/post-title/index.native.js

```diff
@@ -0,0 +1,84 @@
+/**
+ * External dependencies
+ */
+import { TextInput } from 'react-native';
+
+/**
+ * WordPress dependencies
+ */
+import { Component } from '@wordpress/element';
+import { decodeEntities } from '@wordpress/html-entities';
+import { withDispatch } from '@wordpress/data';
+import { withFocusOutside } from '@wordpress/components';
+import { withInstanceId, compose } from '@wordpress/compose';
+
+class PostTitle extends Component {
+	constructor() {
+		super( ...arguments );
+
+		this.onChange = this.onChange.bind( this );
+		this.onSelect = this.onSelect.bind( this );
+		this.onUnselect = this.onUnselect.bind( this );
+
+		this.state = {
+			isSelected: false,
+		};
+	}
+
+	handleFocusOutside() {
+		this.onUnselect();
+	}
+
+	onSelect() {
+		this.setState( { isSelected: true } );
+		this.props.clearSelectedBlock();
+	}
+
+	onUnselect() {
+		this.setState( { isSelected: false } );
+	}
+
+	onChange( title ) {
+		this.props.onUpdate( title );
+	}
+
+	render() {
+		const {
+			placeholder,
+			style,
+			title,
+		} = this.props;
+
+		const decodedPlaceholder = decodeEntities( placeholder );
+
+		return (
+			<TextInput
+				blurOnSubmit={ true }
+				textAlignVertical="top"
+				multiline
+				numberOfLines={ 0 }
+				onChangeText={ this.onChange }
+				onFocus={ this.onSelect }
+				placeholder={ decodedPlaceholder }
+				style={ style }
+				value={ title }>
+			</TextInput>
+		);
+	}
+}
+
+const applyWithDispatch = withDispatch( ( dispatch ) => {
+	const {
+		clearSelectedBlock,
+	} = dispatch( 'core/editor' );
+
+	return {
+		clearSelectedBlock,
+	};
```

Comment:
so this was the thing I was referring to by disptaching to store, just like this one we can also inject onUpdate to props and use editPost( { title } ) to update the post in the store, just like the web side here: https://github.com/WordPress/gutenberg/blob/c2f1059b10ebb3c5457ae61adc05920af4729d58/packages/editor/src/components/post-title/index.js#L180